gdxsv/lbs_discord.go

```diff
@@ -0,0 +1,494 @@
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"math"
+	"net/http"
+	"regexp"
+	"sort"
+	"strconv"
+	"strings"
+	"sync/atomic"
+	"time"
+
+	"go.uber.org/zap"
+)
+
+var retryTimer *time.Timer
+var jobIsRunning uint32
+
+type discordEmbedFooter struct {
+	Text string `json:"text,omitempty"`
+}
+
+type discordEmbedField struct {
+	Name   string `json:"name,omitempty"`
+	Value  string `json:"value,omitempty"`
+	Inline bool   `json:"inline,omitempty"`
+}
+
+type discordEmbed struct {
+	Title       string               `json:"title,omitempty"`
+	Description string               `json:"description,omitempty"`
+	Color       int                  `json:"color,omitempty"`
+	Footer      *discordEmbedFooter  `json:"footer,omitempty"`
+	Timestamp   string               `json:"timestamp,omitempty"`
+	Fields      []*discordEmbedField `json:"fields,omitempty"`
+}
+
+type statusPayload struct {
+	Embed   []*discordEmbed `json:"embeds"`
+	BotName string          `json:"username"`
+}
+
+type lobbyPeers struct {
+	Count          int
+	Name           string
+	RegionName     string
+	Comment        string
+	RenpoPeers     string
+	ZeonPeers      string
+	RenpoRoomPeers string
+	ZeonRoomPeers  string
+	NoForcePeers   string
+}
```

Comment:
I named 'Peer' to mean the end of the connection.
'User' is better to use here.

Reply:
All `Peer` is replaced with `User` now
```go
	type lobbyUsers struct {
		Count                int
		Name                 string
		RegionName           string
		Comment              string
		RenpoUsersString     string
		ZeonUsersString      string
		RenpoRoomUsersString string
		ZeonRoomUsersString  string
		NoForceUsersString   string
	}

	type battleUsers struct {
		RegionName       string
		RenpoUsersString string
		ZeonUsersString  string
	}
```